pkg/storegateway/chunkscache/cache.go

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package chunkscache
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	gokit_log "github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/cache"
+	"github.com/oklog/ulid"
+	"github.com/opentracing/opentracing-go/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/prometheus/tsdb/chunks"
+
+	"github.com/grafana/mimir/pkg/util/spanlogger"
+)
+
+// Range is a contiguous range of chunks. Start is the ref of the first chunk and NumChunks is how many chunks
+// there are in the range.
+type Range struct {
+	BlockID   ulid.ULID
+	Start     chunks.ChunkRef
+	NumChunks int
+}
+
+type ChunksCache interface {
+	FetchMultiChunks(ctx context.Context, userID string, ranges []Range) (hits map[Range][]byte)
+	StoreChunks(ctx context.Context, userID string, r Range, v []byte)
+}
+
+type TracingCache struct {
+	c ChunksCache
+	l gokit_log.Logger
+}
+
+func NewTracingCache(c ChunksCache, l gokit_log.Logger) TracingCache {
+	return TracingCache{
+		c: c,
+		l: l,
+	}
+}
+
+func (c TracingCache) FetchMultiChunks(ctx context.Context, userID string, ranges []Range) (hits map[Range][]byte) {
+	l := spanlogger.FromContext(ctx, c.l)
+	hits = c.c.FetchMultiChunks(ctx, userID, ranges)
+
+	l.LogFields(
```

Comment:
This seems unnecessary? This could be done using the go-kit logging interface with the same effect unless I'm missing something.
```suggestion
  level.Debug(l).Log(
    "name", "ChunksCache.FetchMultiChunks",
    "ranges_requested", len(ranges),
    // ...
  )
```

Reply:
You're right, nice catch